channeldb/db.go

```diff
@@ -288,6 +290,32 @@ func fileExists(path string) bool {
 	return true
 }
 
+// FetchOpenChannel starts a new database transaction and returns the stored
+// active/open channel associated with the target nodeID and channel ID. In the
+// case that there is no channel matching this criteria, an error is returned.
+// In the case that we fail to fetch any open channels, an error is returned.
+func (d *DB) FetchOpenChannel(
+	nodeID *btcec.PublicKey, chanID lnwire.ShortChannelID) (*OpenChannel, error) {
+
+	channels, err := d.FetchOpenChannels(nodeID)
```

Comment:
Instead this should be done by fetching the channel directly from the bucket, instead of iterating over all the node's channels. Not sure if this is compatible with the `shortChanID`, so might have to pass in the `chanPoint` instead,

Reply:
(probably) fixed. now we iterate over a node's `chainhash` buckets, but I think that is acceptable?

Reply:
fixed, we still iterate over the `chainhash` buckets for a node but I think that's acceptable?